ppr-ui/src/utils/registration-helper.ts

```diff
@@ -250,13 +235,12 @@ export async function setupFinancingStatementDraft (stateModel:StateModelIF, doc
   if (draft.financingStatement.vehicleCollateral) {
     stateModel.registration.collateral.vehicleCollateral = draft.financingStatement.vehicleCollateral
   }
-  if (draft.financingStatement.generalCollateral && draft.financingStatement.generalCollateral.length > 0) {
-    stateModel.registration.collateral.generalCollateral = draft.financingStatement.generalCollateral[0].description
+  if (draft.financingStatement.generalCollateral) {
+    stateModel.registration.collateral.generalCollateral = draft.financingStatement.generalCollateral
   }
-  stateModel.registration.collateral.valid = ((stateModel.registration.collateral.vehicleCollateral &&
-      stateModel.registration.collateral.vehicleCollateral.length > 0) ||
-      (stateModel.registration.collateral.generalCollateral &&
-       stateModel.registration.collateral.generalCollateral !== ''))
+  const vcValid = stateModel.registration.collateral.vehicleCollateral?.length !== 0
+  const gcValid = stateModel.registration.collateral.generalCollateral?.length !== 0
```

Comment:
question: is it possible to have general collateral array with a blank general collateral? I guess I just want to make sure the validation is still in place now that we've switched it to an array

Reply:
Oh I see what you mean. With my newer changes I will make sure that never happens with the way it is added (i.e. it will only add when description text is added and if that same description text is deleted fully it will remove that gc from the list

Reply:
coooooooool